src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useCallback, useState, useMemo } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoStatus } from '../../types/TodoStatus';
+
+import { TodoInfo } from '../TodoInfo/TodoInfo';
+import { TodoFilter } from '../TodoFilter/TodoFilter';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  isAdding: boolean;
+  onDelete: (id: number) => void;
+  deletingIds: number[];
+  onChangeStatus: (todo: Todo) => void;
+  onChangeTitle: (id: number, newTitle: string) => void;
+};
+
+export const TodoList: React.FC<Props> = React.memo(({
+  todos,
+  tempTodo,
+  isAdding,
+  onDelete,
+  deletingIds,
+  onChangeStatus,
+  onChangeTitle,
+}) => {
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.ALL);
+
+  const filterTodos = useCallback(() => {
+    let filtered;
+
+    switch (status) {
+      case TodoStatus.ACTIVE:
+      case TodoStatus.COMPLETED:
+        filtered = todos.filter(todo => (
+          (status === TodoStatus.ACTIVE)
+            ? !todo.completed
+            : todo.completed
+        ));
+        break;
+      default:
+        filtered = todos;
+    }
+
+    return filtered;
+  }, [todos, status]);
+
+  const filteredTodos = useMemo(() => filterTodos(), [todos, status]);
```

Comment:
overkill
just use `useMemo` for filtering result 🙂

Reply:
agree